modules/nw-metallb-bgpadvertisement-cr.adoc

```diff
@@ -0,0 +1,73 @@
+
+:_content-type: CONCEPT
+// Module included in the following assemblies:
+//
+// * networking/metallb/metallb-configure-address-pools.adoc
+[id="nw-metallb-bgpadvertisement-cr_{context}"]
+= About the bgpAdvertisement custom resource
+
+The fields for the `bgpAdvertisements` object are defined in the following table:
+
+.BGP advertisements configuration
+[cols="1,1,3a", options="header"]
+|===
+
+|Field
+|Type
+|Description
+
+|`metadata.name`
+|`string`
+|Specifies the name for the BGP advertisement.
+
+|`metadata.namespace`
+|`string`
+|Specifies the namespace for the BGP advertisement.
+Specify the same namespace that the MetalLB Operator uses.
+
+|`spec.aggregationLength`
+|`integer`
+|Optional: Specifies the number of bits to include in a 32-bit CIDR mask.
+To aggregate the routes that the speaker advertises to BGP peers, the mask is applied to the routes for several service IP addresses and the speaker advertises the aggregated route.
+For example, with an aggregation length of `24`, the speaker can aggregate several `10.0.1.x/32` service IP addresses and advertise a single `10.0.1.0/24` route.
+
+|`spec.aggregationLengthV6`
+|`integer`
+|Optional: Specifies the number of bits to include in a 128-bit CIDR mask.
+For example, with an aggregation length of `124`, the speaker can aggregate several `fc00:f853:0ccd:e799::x/128` service IP addresses and advertise a single `fc00:f853:0ccd:e799::0/124` route.
+
+|`spec.communities`
+|`string`
+|Optional: Specifies one or more BGP communities.
+Each community is specified as two 16-bit values separated by the colon character.
+Well-known communities must be specified as 16-bit values:
+
+* `NO_EXPORT`: `65535:65281`
+* `NO_ADVERTISE`: `65535:65282`
+* `NO_EXPORT_SUBCONFED`: `65535:65283`
+
+|`spec.localPref`
+|`integer`
+|Optional: Specifies the local preference for this advertisement.
+This BGP attribute applies to BGP sessions within the Autonomous System.
+
+|`spec.ipAddressPools`
+|`string`
+|Optional: The list of IPAddressPools to advertise with this advertisement, selected by name.
+
+|`spec.ipAddressPoolSelectors`
+|`string`
+|Optional: A selector for the IPAddressPools that gets advertised with this advertisement. If no IPAddressPool is selected by this or by the list, the advertisement is applied to all the IPAddressPools.
```

Comment:
This is for associating ip address pool to advertisement based on the label assigned to ip address pool instead of name itself.